api/src/pcapi/scripts/push_invoices_prod/main.py

```diff
@@ -0,0 +1,1304 @@
+            "Terms": {"value": "30J"},
+            "Type": {"value": "ADR" if is_debit_note else "INV"},
+            "Vendor": {"value": str(invoice.bankAccountId)},
+            "VendorRef": {"value": invoice.reference},
+        }
+
+        response = self._request("PUT", url, params=params, json=body)
+
+        if (
+            response.status_code == 500
+            and self._get_exception_type(response) == "PX.Api.ContractBased.OutcomeEntityHasErrorsException"
+        ):
+            # If the invoice is already pushed, we might be in a retry due to a timeout from the first PUT query
+            # We thus get the id necessary for the POST
+            logger.warning("Invoice '%s' already pushed", invoice.reference)
+            response_json = self.get_invoice(invoice.reference)
+
+        elif response.status_code == 422:
+            logger.warning("Invoice %s has problem with LocationId", invoice.reference)
+            body["LocationID"] = {"value": "MAIN"}
+            response = self._request("PUT", url, params=params, json=body)
+            if response.status_code != 200:
+                raise exceptions.FinanceBackendBadRequest(response, "Error in invoice creation payload")
+
+            else:
+                response_json = response.json()
+
+        elif response.status_code != 200:
+            raise exceptions.FinanceBackendBadRequest(response, "Error in invoice creation payload")
+
+        else:
+            response_json = response.json()
+
+        if response_json.get("Status", {}).get("value") == "Balanced":
+            # Set invoice to Open in Cegid
+            set_open_url = f"{self.base_url}/{self._interface}/Bill/ReleaseBill"
+            set_open_response = self._request("POST", set_open_url, json={"Entity": {"id": response_json["id"]}})
+
+            if set_open_response.status_code // 100 != 2:
+                raise exceptions.FinanceBackendBadRequest(set_open_response, "Error in setting invoice to Open")
+
+        return response_json
+
+
+def _push_invoices(count: int) -> None:
+    if bool(app.redis_client.exists(conf.REDIS_PUSH_INVOICE_LOCK)):
+        return
+
+    invoices_query = (
+        db.session.query(finance_models.Invoice)
+        .filter(
+            sa.or_(
+                finance_models.Invoice.status == finance_models.InvoiceStatus.PENDING,
+                sa.and_(
+                    finance_models.Invoice.status == finance_models.InvoiceStatus.PAID,
+                    finance_models.Invoice.reference.in_(INVOICES_NOT_IN_CEGID_YET),
+                ),
+            )
+        )
+        .with_entities(finance_models.Invoice.id)
+    )
+    if count != 0:
+        invoices_query = invoices_query.limit(count)
+
+    invoices = invoices_query.all()
+
+    if not invoices:
+        return
+
+    app.redis_client.set(conf.REDIS_PUSH_INVOICE_LOCK, "1", ex=conf.REDIS_PUSH_INVOICE_LOCK_TIMEOUT)
+
+    try:
+        invoice_ids = [e[0] for e in invoices]
```

Comment:
pour éviter les doublons je mettrais : 
```suggestion
        invoice_ids = {e[0] for e in invoices}
```